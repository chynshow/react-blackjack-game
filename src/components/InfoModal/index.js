import React, { useContext } from 'react';
import { AppContext } from '../../state/AppContext';
import { HIDE_INFO_MODAL } from '../../state/AppReducer';

const InfoModal = () => {
  const { infoModal, dispatch } = useContext(AppContext);
  const handelOnClick = () => {
    if (!infoModal?.cb) return dispatch({ type: HIDE_INFO_MODAL });
    infoModal?.cb();
  };

  return (
    <>
      <div
        className={`${
          infoModal?.isActive
            ? 'c-info-modal c-info-modal--active'
            : 'c-info-modal'
        }`}
      >
        <h5 className='c-info-modal__title'>{infoModal?.title}</h5>
        <div className='c-info-modal__msg'>{infoModal?.msg}</div>

        <button className='c-btn c-info-modal__btn' onClick={handelOnClick}>
          {infoModal?.closeBtnTitle}
        </button>
      </div>
      {infoModal?.isActive && <div className='c-info-modal__overlay' />}
    </>
  );
};

export default InfoModal;
