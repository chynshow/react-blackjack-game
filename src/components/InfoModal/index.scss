.c-info-modal {
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 50vw;
  height: 50vw;
  border-radius: 5px;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  color: #333;
  background: #fafafa;
  transition: 0.2s ease-in-out;
  transform: scale(0);
  visibility: hidden;
  z-index: 2;

  &__title {
    font-family: 'Montserrat', sans-serif;
    font-size: 2rem;
  }

  &__msg {
  }

  &__btn {
    background: #333;
    color: #fafafa;
  }

  &__overlay {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: #333;
    opacity: 0.2;
    z-index: 1;
  }

  &--active {
    transform: scale(1);
    visibility: visible;
  }
}
