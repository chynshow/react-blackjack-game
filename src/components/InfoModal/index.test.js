import React from 'react';
import InfoModal from './index';
import { shallow } from 'enzyme';

it('render InfoModal component', () => {
  const c = shallow(<InfoModal />);
  expect(c).toMatchSnapshot();
});
