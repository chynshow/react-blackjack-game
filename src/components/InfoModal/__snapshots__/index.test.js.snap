// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render InfoModal component 1`] = `
<Fragment>
  <div
    className="c-info-modal"
  >
    <h5
      className="c-info-modal__title"
    />
    <div
      className="c-info-modal__msg"
    />
    <button
      className="c-btn c-info-modal__btn"
      onClick={[Function]}
    />
  </div>
</Fragment>
`;
